gordo_components/builder/build_model.py

```diff
@@ -143,6 +141,50 @@ def _save_model_for_workflow(
     return output_dir
 
 
+def _get_metadata(model: BaseEstimator, metadata: dict = dict()) -> dict:
+    """
+    Recursively check for :class:`gordo_components.model.base.GordoBase` in a
+    given ``model``. If such a model exists buried inside of a
+    :class:`sklearn.pipelilne.Pipeline` which is then part of another
+    :class:`sklearn.base.BaseEstimator`, this function will return its metadata.
+
+    Parameters
+    ----------
+    model: BaseEstimator
+    metadata: dict
+        Any initial starting metadata, but is mainly meant to be used during
+        the recursive calls to accumulate any multiple :class:`gordo_components.model.base.GordoBase`
+        models found in this model
+
+    Notes
+    -----
+    If there is a ``GordoBase`` model inside of a ``Pipeline`` which is not the final
+    step, this function will not find it.
+
+    Returns
+    -------
+    dict
+        Dictionary representing accumulated calls to :meth:`gordo_components.model.base.GordoBase.get_metadata`
+    """
+    metadata = metadata.copy()
+
+    if not isinstance(model, GordoBase):
+
+        for attr, val in model.__dict__.items():
+            if isinstance(val, Pipeline):
+                final_step = _get_final_gordo_base_step(val)
```

Comment:
Oh, my head explodes. This _get_final_gordo_base_step is also recursive.

Reply:
Removed the function entirely, because, as you pointed out, the final step should actually be a BaseEstimator which in turn took a `GordoBase` as an estimator. :+1: